Admin documentation (#633)
SwissDataScienceCenter/renku: docs/admin/prerequisites/certificates.rst

.. _certificates:

Certificates
============

> You need a certificate for the ingress, you can deploy a certificate manager like LetsEncrypt (A.) or create manually a certificate and instert it as a k8s secret (B.).

Reviewer: ```suggestion
To use TLS you need valid certificates for the ingress by either deploying a certificate manager like LetsEncrypt or creating certificates manually and inserting them as k8s secrets.
```

> A. Use LetsEncrypt

Reviewer: ```suggestion
LetsEncrypt
```

------------------

Copy and edit `manifests/cert-manager-issuer.yaml file <https://github.com/SwissDataScienceCenter/renku-admin-docs/blob/master/manifests/cert-manager-issuer.yaml>`_ to fill in the `email` field.

.. code-block:: console

   $ helm upgrade --install cert-manager stable/cert-manager -f helm-installs/cert-manager-values.yaml --namespace kube-system
   $ kubectl apply -f manifests/cert-manager-issuer.yaml

Note: recent cert-manager versions require a trick to get deployed, the workaround is to execute the following command before installing cert-manager:

.. code-block:: console

   $ helm install --version 0.4.1 --name cert-manager stable/cert-manager -f helm-installs/cert-manager-values.yaml --namespace kube-system

Check that we can issue certificates automatically.

[TODO add simple deployment to test certificates and steps to do it]

> B. Generate manually

Reviewer: ```suggestion
Generate manually
```

--------------------

[TODO generate certificate crt]

> Add the previously created certificate as a secret to Renku namespace.

Reviewer: ```suggestion
Add the previously created certificate as a secret to the ``renku`` namespace.
```
